src/main/java/ch/njol/skript/expressions/ExprSignText.java

```diff
@@ -32,145 +26,197 @@
 import ch.njol.skript.doc.Examples;
 import ch.njol.skript.doc.Name;
 import ch.njol.skript.doc.Since;
-import ch.njol.skript.effects.Delay;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
 import ch.njol.skript.lang.util.SimpleExpression;
 import ch.njol.skript.lang.util.SimpleLiteral;
+import ch.njol.skript.util.Utils;
 import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+
+import org.bukkit.block.Block;
+import org.bukkit.block.Sign;
+import org.bukkit.event.Event;
+import org.bukkit.event.block.SignChangeEvent;
+import org.eclipse.jdt.annotation.Nullable;
 
-/**
- * @author Peter Güttinger
- */
 @Name("Sign Text")
 @Description("A line of text on a sign. Can be changed, but remember that there is a 16 character limit per line (including color codes that use 2 characters each).")
-@Examples({"on rightclick on sign:",
-		"	line 2 of the clicked block is \"[Heal]\":",
-		"		heal the player",
-		"	set line 3 to \"%player%\""})
+@Examples({
+	"on rightclick on sign:",
+		"\tline 2 of the clicked block is \"[Heal]\":",
+		"\t\theal the player",
+		"\tset line 3 of block at {_sign} to \"%player%\""
+})
 @Since("1.3")
 public class ExprSignText extends SimpleExpression<String> {
+
 	static {
 		Skript.registerExpression(ExprSignText.class, String.class, ExpressionType.PROPERTY,
-				"[the] line %number% [of %block%]", "[the] (1¦1st|1¦first|2¦2nd|2¦second|3¦3rd|3¦third|4¦4th|4¦fourth) line [of %block%]");
+				"[the] lines [of %block%]",
+				"[the] line %number% [of %block%]",
+				"[the] (1¦1st|1¦first|2¦2nd|2¦second|3¦3rd|3¦third|4¦4th|4¦fourth) line [of %block%]"
+			);
```

Comment:
could the last pattern now be `(1:(1st|first)|2:(2nd|second)|3:(3rd|third)|4:(4th|fourth))` for the matching?
we no longer need pipe lines and should stay back from those as ParseMark supports colons.

Do double check with the pattern tho but it should work like that